### ISSUE TITLE
Question about loss

### ISSUE THREAD

**Issue by hw-choo** [WenjieDu/SAITS#5]
Thank you for your work and please understand that it is not a direct question about the code. Is there any reason the loss function does not include the classification error term? Some models that perform reconstruction and imputation are include classification error in the loss function. Have you ever trained models in this way? If so, please let me know what the results were like.

**Comment by WenjieDu** [comment]
Hi there,<br><br>Thank you so much for your attention to SAITS! If you find SAITS is helpful to your work, please star⭐️ this repository. Your star is your recognition, which can let others notice SAITS. It matters and is definitely a kind of contribution.<br><br>I have received your message and will respond ASAP. Thank you again for your patience! 😃<br><br>Best,<br>Wenjie

**Comment by WenjieDu** [comment]
Thank you for raising this issue to discuss. You can append a classification task to train SAITS (just like BRITS does) if your work needs it. Actually, I did think about this when I was designing SAITS. We can extend the jointly-optimization training approach into a multi-task one including classification. But finally, I didn't, and the reasons are listed below. 

Firstly, as the title illustrated, this paper focuses mainly on the imputation problem. So we try to solve it well to surpass the state-of-the-art. As you can see in the experiment 'Downstream Classification Task', our SAITS with a two-stage method (impute then classify) achieves comparable performance to the end-to-end model BRITS in [^1]. This qualitative evaluation together with the quantitative imputation accuracy evaluation in the paper demonstrates the quality of imputation from SAITS.

Secondly, as I said above, if one's work needs SAITS to perform end-to-end classification, one can add additional NN modules and append the classification task in the training to see its performance. This is very intuitive and natural, and cannot be claimed as a novelty in the paper. Different from ORT+MIT utilized to solve the optimization problem of imputation for self-attention models, appending a classification task here is more like applying SAITS structure to solve a classification task, this is an application. Also, the classification task may disturb the optimization of the imputation problem (in my model training, I observed imputation accuracy degraded). Therefore, if we append the classification task, it will be kind of redundant. 

If you are interested in this topic, you can give it a try. And if you can report your investigation here, I can pin this issue to let others notice your work and continue the discussion. This won't be a problem.


[^1]: Cao, W., Wang, D., Li, J., Zhou, H., Li, L., & Li, Y. (2018). BRITS: Bidirectional Recurrent Imputation for Time Series. NeurIPS 2018.

**Comment by hw-choo** [comment]
Thank you for your kind reply. I will try and share it if there are interesting results.

**Comment by xuzengsong** [comment]
Thank you for your work，I recently read your paper SAITS: Self-Attention-based Imputation for Time Series. I am also doing the work related to multivariate time series Imputation. I have some questions, and I hope to communicate with you.
  1.I recently used your method to run the data set I used. My data processing approach is first divided into training set and test set, and then build time sequence, first use the train set to train, and then use the test set test (but I know data Imputation algorithm is unsupervised algorithm and did not use the true information of the missing data, there are some people who divided the test set training set, while there are also some people didn't,There are some differences in the results of your algorithm between these two data set partitioning methods,May I ask how do you view the partitioning of data sets?)
  2. May I ask whether your algorithm will have overfitting, because the loss of back propagation is the MAE of unmissing items, not the MAE of the whole data set. I feel that with the increase of training times, it will gradually tend to be overfitting
  3. Now the stopping condition of the algorithm is to reach the specified epoch. The epoch of different data sets need to be detected,If we divide the test set and the training set, can we quit the training by judging that the missing item data MAE of the training set reaches the minimum.
Thank you very much
